### PR TITLE
Update GH Actions workflow, add unit testing for input errors, and other miscellaneous changes

### PR DESCRIPTION
**Pull request by HaydenMacDonald**
I've made several changes to this package which I've documented below. Please let me know your thoughts! Note that I've added `\dontrun{}` tags around examples that are currently problematic (e.g. they return errors *sometimes*). 

Github Actions workflow
- Add support for multiple OSs

Testing
- Introduce tests for each function to boost codecov score to ~100%
- Add `\dontrun{}` tags on particular examples that were failing

Miscellaneous
- Chang `.onLoad` function to `.onAttach` to avoid RCMDCHECK notes
- Import .data from `rlang` to avoid notes associated with `tidy_select()`
- Add `as_tibble()` to `opr()` with default values

### CODE REVIEW

**Comment by HaydenMacDonald** [comment]
Hey, @philip-khor! Let me know if you have any feedback regarding this pull request. I am happy to amend some of these changes if they don't fit with the original vision for the package. Thanks!

**Comment by philip-khor** [comment]
Hi @HaydenMacDonald, sorry I missed this! Thanks for contributing, weirdly enough the diffs are pretty huge, but from the looks of it these look like minor changes. I'll try to get through this ASAP :)

File: R/currencies.R [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
+#' forward prices for trade settlement and RMB deposit acceptance rates
+#' from the BNM API as a tidy tibble.
+#' @param type String. Either "dar" or "fx_forward".
+#' @keywords api
+#' @importFrom tidyr spread unnest
+#' @importFrom tibble tibble
+#' @importFrom purrr map_dbl
+#' @importFrom rlang .data
+#' @examples
+#' \dontrun{renminbi(type = "dar")}
```
Comment (philip-khor) [inline R/currencies.R:13]: LGTM, thanks for this addition! Afaik the API returns errors when there's too many calls, I've been meaning to write in to see if they'd up the limit for requests from this package, but I don't think the usage of this package is high enough to justify :)

File: R/monthly.R [hunk 2]
```diff
@@ -1,234 +1,234 @@
...
+          Sys.sleep(1)
+          flatten(get_bnm_tbl(glue("{stub}/year/{year}/month/{x}"),
+                              query = args))
+            }
+          )
+      }
+
+
+    } else if (!is.null(year)) {
+      stopifnot(is.numeric(year) & is.numeric(month))
```
Comment (philip-khor) [inline R/monthly.R:29]:
```suggestion
      stopifnot(is.numeric(year) && is.numeric(month))

```

Comment (philip-khor) [reply R/monthly.R:29]: we should expect year and month of length 1 here, I was trying to extend the functionality to accept vectors but from line 30 I didn't quite get there for this function

**Review by philip-khor — CHANGES_REQUESTED** [review]
Great stuff, thanks! Minor nitpick:

**Review by philip-khor — APPROVED** [review]

**Comment by HaydenMacDonald** [comment]
Cheers! Happy to contribute 🙂

**Comment by philip-khor** [comment]
Btw can I check why the lubridate package was needed in this case? I don't recall having any dependencies on lubridate in the package functions ... would prefer to import just the functions we need rather than the entire package, in which case 'suggests' would be more suitable.

**Comment by HaydenMacDonald** [comment]
I checked through the tests and R files. The lubridate package was likely part of a change that didn't make it to this pull request. Feel free to remove it!

**Comment by philip-khor** [comment]
Thanks for clarifying!

**Pull request merged**